source/opt/inline_pass.cpp

```diff
@@ -749,6 +1062,85 @@ void InlinePass::InitializeInline() {
     // Compute inlinability
     if (IsInlinableFunction(&fn)) inlinable_.insert(fn.result_id());
   }
+
+  if (get_module()->ContainsOpenCL100DebugInstrunctions()) {
```

Comment:
I agree, but I am not sure this will be a general analysis or not. For the `DebugInlinedAt` information, I do not think it will be reused but we may reuse `param_id2debugdecl_`, `id2lexical_scope_`, and `func_id2dbg_func_id_`.

I want to delay creating an analysis for it until we actually reuse it. What do you think?

Reply:
I would prefer to do it right away.  It does a couple good things.  First it make the inlining code simpler because it does not include this analysis code.  Keeping code modular is good for readability.

The second part is that it make it more discoverable.  Why would someone look inside the inlining code for this?  If someone else want to use it later, they are less likely to rewrite it themselves.